docs/definition.rst

```diff
@@ -290,3 +334,85 @@ Each list item must be a dictionary containing the following (non-optional) keys
   build context directory that should contain the source file(s) (e.g., ``files/configs``).
   This may not be an absolute path or contain ``..`` within the path. This directory
   will be created for you if it does not exist.
+
+.. _dependencies_v2:
+
+dependencies (v2)
+^^^^^^^^^^^^^^^^^
+
+The ``dependencies`` section for version 2 is similar to its
+:ref:`version 1 counterpart <dependencies_v1>`. The exception is that three new
+keywords are added, and the values of the existing ``galaxy``, ``python`` and
+``system`` keys may either be the name of a file, or inline representations
+of those files. The ``python`` and ``system`` values can be a list of dependency
+values, but the ``galaxy`` value must be a string representation of the Galaxy
+requirements YAML.
+
+For example, this format is supported in both versions:
+
+.. code:: yaml
+
+    dependencies:
+        python: requirements.txt
+        system: bindep.txt
+        galaxy: requirements.yml
+
+And this format, only supported in version 2, uses inline values:
+
+.. code:: yaml
+
+    dependencies:
+        python:
+          - pywinrm
+        system:
+          - iputils [platform:rpm]
+        galaxy: |
```

Comment:
This could probably use an explanatory comment for folks that don't know what `|` does in a YAML document;  like "the `|` symbol causes all following lines at this indent level or higher to be included as a literal string" or something?

Reply:
👍🏻